mikrotik/resource_bgp_peer.go

```diff
@@ -0,0 +1,318 @@
+			},
+			"remote_as": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"remote_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"instance": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"address_families": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "ip",
+			},
+			"ttl": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "default",
+			},
+			"default_originate": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "never",
+			},
+			"hold_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "3m",
+			},
+			"nexthop_choice": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "default",
+			},
+			"out_filter": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"in_filter": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"allow_as_in": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"as_override": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"cisco_vpls_nlri_len_fmt": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"comment": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"disabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"keepalive_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"max_prefix_limit": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"max_prefix_restart_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"multihop": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"passive": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"remote_port": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"remove_private_as": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"route_reflect": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"tcp_md5_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"update_source": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"use_bfd": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceBgpPeerCreate(d *schema.ResourceData, m interface{}) error {
+	instance := prepareBgpPeer(d)
+
+	c := m.(client.Mikrotik)
+
+	bgpPeer, err := c.AddBgpPeer(instance)
```

Comment:
nit: "instance" is slightly confusing in this context since there is a bpg instance resource. Why don't we call this `peer` instead?